hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,749 @@
+        loaded_nc_files = read_nc(
+            fn, self.root, logger=logger, single_var_as_array=False, **kwargs
+        )  # TODO: decide where read_nc should be placed
+        for ds in loaded_nc_files.values():
+            self.set(ds)
+
+    def create(self):
+        """Create grid."""
+        raise NotImplementedError
+
+    @property
+    def res(self) -> Tuple[float, float] | None:
+        """Returns the resolution of the model grid."""
+        if len(self._data) > 0:
+            return self._data.raster.res
+        _exec_nodata_strat(
+            msg="No grid data found for deriving resolution",
+            strategy=NoDataStrategy.IGNORE,
+            logger=logger,
+        )
+
+    @property
+    def transform(self) -> Affine | None:
+        """Returns spatial transform of the model grid."""
+        if len(self._data) > 0:
+            return self._data.raster.transform
+        _exec_nodata_strat(
+            msg="No grid data found for deriving transform",
+            strategy=NoDataStrategy.IGNORE,
+            logger=logger,
+        )
+
+    @property
+    def crs(self) -> CRS | None:
+        """Returns coordinate reference system embedded in the model grid."""
+        if self._data.raster.crs is not None:
+            return CRS(self._data.raster.crs)
+        logger.warn("Grid data has no crs")
+
+    @property
+    def bounds(self) -> List[float] | None:
+        """Returns the bounding box of the model grid."""
+        if len(self._data) > 0:
+            return self._data.raster.bounds
+        _exec_nodata_strat(
+            msg="No grid data found for deriving bounds",
+            strategy=NoDataStrategy.IGNORE,
+            logger=logger,
+        )
+
+    @property
+    def region(self) -> gpd.GeoDataFrame | None:
+        """Returns the geometry of the model area of interest."""
+        if len(self._data) > 0:
+            crs = self.crs
+            if crs is not None and hasattr(crs, "to_epsg"):
+                crs = crs.to_epsg()  # not all CRS have an EPSG code
+            return gpd.GeoDataFrame(geometry=[box(*self.bounds)], crs=crs)
+        _exec_nodata_strat(
+            msg="No grid data found for deriving region",
+            strategy=NoDataStrategy.IGNORE,
+            logger=logger,
+        )
+
+    @property
+    def grid(self) -> xr.Dataset:
+        """Model static gridded data as xarray.Dataset."""
+        if self._data is None:
+            self._initialize_grid()
+        return self._data
+
+    def _initialize_grid(self, skip_read: bool = False, read: bool = True) -> None:
+        """Initialize grid object."""
+        if self._data is None:
+            self._data = xr.Dataset()
+            if read and not skip_read:
+                self.read()
+
+    def set_crs(self, crs: CRS) -> None:
+        """Set coordinate reference system of the model grid."""
+        if len(self._data) > 0:
+            self._data.raster.set_crs(crs)
+
+    def setup_grid_from_constant(
```

Comment:
We discussed in a new issue we wanted to change these names. Here we could rename to add_data_from_constant